scripts/Diffraction/single_crystal/base_sx.py

```diff
@@ -614,6 +618,180 @@ def remove_non_integrated_peaks(peaks):
+        imax = np.argmax(radii)
+        # calc center in frame of ellipsoid axes
+        cen = getattr(pk, frame_to_peak_centre_attr)()
+        cen = np.matmul(evecs.T, np.array(cen) + translation)
+        # get extents
+        box_lengths = bg_outer_radii if not np.allclose(bg_outer_radii, 0.0) else radii
+        box_lengths = box_lengths * extent
+        extents = np.vstack((cen - box_lengths, cen + box_lengths))
+        # get nbins along each axis
+        nbins = np.array([int(nbins_max * radii[iax] / radii[imax]) for iax in range(ndims)])
+        # ensure minimum of 3 bins inside radius along each dim
+        min_nbins_in_radius = 3
+        nbins_in_radius = np.min(nbins * radii / box_lengths)  # along most coarsely binned dimension
+        if nbins_in_radius < min_nbins_in_radius:
+            nbins = nbins * min_nbins_in_radius / nbins_in_radius
+        # call BinMD
+        ws_cut = mantid.BinMD(
+            InputWorkspace=wsMD,
+            OutputWorkspace="__ws_cut",
+            AxisAligned=False,
+            BasisVector0=r"Q$_0$,unit," + ",".join(np.array2string(evecs[:, 0], precision=6).strip("[]").split()),
+            BasisVector1=r"Q$_1$,unit," + ",".join(np.array2string(evecs[:, 1], precision=6).strip("[]").split()),
+            BasisVector2=r"Q$_2$,unit," + ",".join(np.array2string(evecs[:, 2], precision=6).strip("[]").split()),
+            OutputExtents=extents.flatten(order="F"),
+            OutputBins=nbins.astype(int),
+            EnableLogging=False,
+            StoreInADS=False,
+        )
+        return ws_cut, radii, bg_inner_radii, bg_outer_radii, box_lengths, imax
+
+    @staticmethod
+    def _get_ellipsoid_params_from_peak(peak_shape: PeakShape, ndims: int):
+        shape_info = json_loads(peak_shape.toJSON())
+        if peak_shape.shapeName().lower() == "spherical":
+            BaseSX.convert_spherical_representation_to_ellipsoid(shape_info)
+        # get radii
+        radii = np.array([shape_info[f"radius{iax}"] for iax in range(ndims)])
+        bg_inner_radii = np.array([shape_info[f"background_inner_radius{iax}"] for iax in range(ndims)])
+        bg_outer_radii = np.array([shape_info[f"background_outer_radius{iax}"] for iax in range(ndims)])
+        # eignevectors of ellipsoid
+        evecs = np.zeros((ndims, ndims))
+        for iax in range(ndims):
+            evec = np.array([float(elem) for elem in shape_info[f"direction{iax}"].split()])
+            evecs[:, iax] = evec / np.linalg.norm(evec)
+        # get translation in frame of wsMD
+        translation = np.array([shape_info[f"translation{iax}"] for iax in range(ndims)])
+        return radii, bg_inner_radii, bg_outer_radii, evecs, translation
+
+    @staticmethod
+    def _convert_spherical_representation_to_ellipsoid(shape_info: dict):
+        # copied from mantidqt.widgets.sliceviewer.peaksviewer.representation.ellipsoid - can't import here though
```

Comment:
This might be a suggestion for a future PR, to move this to a common/utility location

Reply:
So I believe the problem is that I can't import things from `mantidqt` outside of `mantidqt` without breaking things (unless that's no longer the case?). Where is the best place for such a utils library/folder?

Reply:
Good question, maybe we want a diffraction folder here? https://github.com/mantidproject/mantid/tree/main/Framework/PythonInterface/mantid/utils

Looks like reflectometry have started using it as a utility directory for their orso helper `from mantid.utils.reflectometry.orso_helper`

Reply:
Actually I'm not so sure, because this is the PythonInterface directory 🤔 Maybe ignore this comment for now!